src/v2/view-builder/views/piv/ChallengePIVView.js

```diff
@@ -0,0 +1,87 @@
+import { loc, createCallout } from 'okta';
+import { BaseForm } from '../../internals';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import PIVHeader from './PIVHeader';
+
+const Body = BaseForm.extend({
+
+  className: 'piv-cac-card',
+  modelEvents: {
+    request: 'startAuthentication',
+    error: 'stopAuthentication',
+  },
+
+  initialize() {
+    BaseForm.prototype.initialize.apply(this, arguments);
+    this.model.set('useRedirect', true);
+    this.addInstructions();
+  },
+
+  title() {
+    return loc('piv.cac.title', 'login');
+  },
+
+  save() {
+    return loc('retry', 'login');
+  },
+
+  addInstructions() {
+    this.add(
+      `<div class='piv-verify-text'>
+        <p>${loc('piv.cac.card.insert', 'login')}</p>
+        <div data-se='piv-waiting' class='okta-waiting-spinner'></div>
+      </div>`
+    );
+  },
+  
+  startAuthentication: function() {
+    this.$('.okta-waiting-spinner').show();
```

Comment:
Using this kind of toggling is a recipe to introduce bugs in future.. Where developer always needs to be aware on when to show and hide..

can you try to leverage triggering an event to show and hide.. Something similar was done by @TianGan-okta for some device view..

Reply:
@TianGan-okta if you can point me to your PR, i can take a look

Reply:
Is this what you are looking for? https://github.com/okta/okta-signin-widget/pull/1602/files#diff-407ec8b6fe1d67d950995d078f8c9c827615326e41069611bfa88133476ba9e4R67

Reply:
This is it.. you rock @TianGan-okta 🎸

Reply:
Filed https://oktainc.atlassian.net/browse/OKTA-406380 for further discussion